src/Profile/EditProfile.tsx

```diff
@@ -57,72 +58,80 @@ const CreateOrEditProfile = (props: Props) => {
         let passwordLengthError: boolean = false;
         let passwordMatchError: boolean = false;
         let phoneNumberError: boolean = false;
-
+        let alphanumericError: boolean = false;
 
         if (!profile.firstName || profile.firstName === "") {
             nameError = true;
         }
         if (!profile.lastName || profile.lastName === "") {
             lastNameError = true;
         }
-        if (!password || password.length<7) {
+        if (!profile.password || profile.password.length<7) {
             passwordLengthError = true;
         }
-        if (password !== confirmPassword) {
+        if (profile.password !== confirmPassword) {
```

Comment:
The user can change other data without changing the password. So you only have to validate the password requirements if the user wants to change the password.

The password validation should be done only when the user enter some text on the `password` or on the `confirm password` field.

Reply:
Stays like this for this version